src/actions/language.actions.ts

```diff
@@ -14,10 +18,24 @@ export const getLanguageLevels = async () => {
   return data;
 };
 
-export const getLanguageTopics = async () => {
+export const getLanguageTopics = async ({
+  langId,
+  userId,
+}: {
+  langId: number;
+  userId: string;
+}) => {
   const supabase = createClient();
 
-  const { data, error } = await supabase.from("languages_topics").select("*");
+  const { data, error } = await supabase
+    .from("languages_topics")
+    .select("*, languages_db(*), languages_quiz(*)")
+    .eq("languages_db.language_id", langId)
+    .eq("languages_quiz.user_id", userId)
+    .eq("languages_quiz.language_id", langId)
```

Comment:
checking language_id twice

Reply:
these are two different tables, languages_db and languages_quiz.